client/src/pages/addFlight/index.js

```diff
@@ -1,48 +1,134 @@
-import React, { Component } from 'react';
+import React, {
+  Component
+} from "react";
 
-import { DurationCard, Select, Input, TimeButton, Button, Header, SideNav} from '../../components'
-import './addflight.css';
+import {
+  DurationCard,
+  Select,
+  Input,
+  TimeButton,
+  Button,
+  Header,
+  SideNav
+} from "../../components";
 
-class UpdateFlight extends Component {
-  constructor(props){
-    super(props)
-    this.state = {
-      selectValue: '',
-    };
+import "./addflight.css";
 
-    this.handleSelectChange = this.handleSelectChange.bind(this);
+class AddFlight extends Component {
+  state = {
+    originValue: "",
+    destinationValue: "",
+    dateValue: "",
+    airCraftTypeValue: ""
+  };
 
-  }
+  handleAddFlight = e => {
+    e.preventDefault();
+    const {
+      originValue,
+      destinationValue,
+      dateValue,
+      airCraftTypeValue
+    } = this.state;
+    const data = JSON.stringify({
+      originValue,
+      destinationValue,
+      dateValue,
+      airCraftTypeValue
+    });
+    fetch("/addFlight_post", {
```

Comment:
One thing that might be useful is to add a file which exports all of your API calls.  Then, inside that file, you can have one helper function which does the fetch, and that's the function that can contain common code for all of your app's fetch calls.
For example, your API.js would have a "addFlight" function which just takes a data object, and it calls your "doRequest" function, which would do the JSON.stringify, always set the `credentials: same-origin`, and in the future once you have login giving you auth tokens, it would set the Authorization header too.  And it could even redirect to the login page on any network request that fails with 401 unauthorized!